src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -39,7 +39,9 @@ public SurfaceNamer(NameFormatter languageNamer) {
     super(languageNamer);
   }
 
-  public static final String NOT_IMPLEMENTED = "$ NOT IMPLEMENTED $";
+  public String getNotImplementedString(String feature) {
```

Comment:
IIRC that discussion was about this exact thing. The argument for failing the generation process immediately is that there would be a stack trace so you could know exactly where it failed; however, with the extra feature information I have added here, that particular aspect is mitigated. I think the full solution is to do a scan of the generated file before exiting the process, and if it contains one of these strings, then fail; that way, the overall output has still been written out and can be inspected.